Source/engine.cpp

```diff
@@ -557,18 +558,44 @@ void SetPixel(const CelOutputBuffer &out, Point position, BYTE col)
 	*out.at(position.x, position.y) = col;
 }
 
-void DrawLineTo(const CelOutputBuffer &out, Point a, Point b, BYTE color_index)
+void DrawHorizontalLine(const CelOutputBuffer &out, Point from, int width, std::uint8_t colorIndex)
 {
-	int dx = b.x - a.x;
-	int dy = b.y - a.y;
-	int steps = abs(dx) > abs(dy) ? abs(dx) : abs(dy);
-	float ix = dx / (float)steps;
-	float iy = dy / (float)steps;
-	float sx = a.x;
-	float sy = a.y;
-
-	for (int i = 0; i <= steps; i++, sx += ix, sy += iy) {
-		SetPixel(out, { static_cast<int>(sx), static_cast<int>(sy) }, color_index);
+	if (from.y < 0 || from.y >= out.h() || from.x >= out.w() || width <= 0 || from.x + width <= 0)
+		return;
+	if (from.x < 0) {
+		width += from.x;
+		from.x = 0;
+	}
+	if (from.x + width > out.w())
+		width = (from.x + width) - out.w();
+	return UnsafeDrawHorizontalLine(out, from, width, colorIndex);
+}
+
+void UnsafeDrawHorizontalLine(const CelOutputBuffer &out, Point from, int width, std::uint8_t colorIndex)
+{
+	std::memset(out.at(from.x, from.y), colorIndex, width);
+}

```

Comment:
There's already pretty much the same code in the QoL features (in common.h/cpp I believe).

Maybe farm out both of these into a common location?